### ISSUE TITLE
[FP] When ``documentInternalElements`` is ``false`` in stylecop.json and it sees ``internal interface`` type.

### ISSUE THREAD

**Issue by AraHaan** [DotNetAnalyzers/StyleCopAnalyzers#3810]
I get warning ``SA1600`` when the following code is seen in a project of mine at least within Visual Studio:

```cs
internal interface IBObject
{
}
```

Expected: No warning

Result: Warning is issued.

**Comment by bjornhellander** [comment]
DocumentInterfaces=true takes precedence over DocumentInternalElements=false, so this will happen if DocumentInterfaces is not set to false. Is that the case for you?

**Comment by AraHaan** [comment]
> DocumentInterfaces=true takes precedence over DocumentInternalElements=false, so this will happen if DocumentInterfaces is not set to false. Is that the case for you?

That is not in the json so I am guessing that it is by default true. I think in that case DocumentInterfaces *should* consider the internal elements though and should skip the internal ones if the ``DocumentInternalElements`` is false though.

**Comment by sharwell** [comment]
Duplicate of #3100

**Comment by AraHaan** [comment]
> The current implementation makes it possible to at least require documentation for interfaces, which seems reasonable to me. If you don't want to treat interfaces specially, then you can set DocumentInterfaces to false.
> 
> Which elements would you like it to require documentation for?

public interfaces only.
